system/filesystem.py

```diff
@@ -101,6 +106,20 @@ def _get_fs_size(fssize_cmd, dev, module):
         # There is no way to get the blocksize and blockcount for btrfs filesystems
         block_size = 1
         block_count = 1
+    elif fssize_cmd == 'swapon':
+        # Old swapon versions support only: swapon -s
+        # which is the same than parsing /proc/swaps;
+        # ie: awk '$1 == "$DEVICE_PATH" {print $3}' /proc/swaps
+        # and in which the size is expressed in kB.
+        # Newer swapon versions should use: swapon --noheadings --show --raw --bytes
+        rc, size, err = module.run_command("%s -s" % (cmd, ))
+        if rc == 0:
+            for line in size.splitlines():
+                row = line.split()
+                if row[0] == dev:
+                    return int(row[2]) * 1024
```

Comment:
I think it would be better to not add a 'return' here when the other part part of the code don't.

And if the swap device is not in us, it will fail, and continue to 'return block_size*block_count' with 2 unassigned variable.